InvenTree/common/api.py

```diff
@@ -130,8 +131,104 @@ class UserSettingsDetail(generics.RetrieveUpdateAPIView):
     ]
 
 
-common_api_urls = [
+class NotificationList(generics.ListAPIView):
+    queryset = common.models.NotificationMessage.objects.all()
+    serializer_class = common.serializers.NotificationMessageSerializer
+
+    filter_backends = [
+        DjangoFilterBackend,
+        filters.SearchFilter,
+        filters.OrderingFilter,
+    ]
+
+    ordering_fields = [
+        'category',
+        'name',
+        'read',
+    ]
+
+    search_fields = [
+        'name',
+        'message',
+    ]
+
+    filter_fields = [
+        'category',
+        'read',
+    ]
+
+    def filter_queryset(self, queryset):
+        """
+        Only list notifications which apply to the current user
+        """
+
+        try:
+            user = self.request.user
+        except AttributeError:
+            return common.models.NotificationMessage.objects.none()
+
+        queryset = super().filter_queryset(queryset)
+        queryset = queryset.filter(user=user)
+        return queryset
+
+
+class NotificationDetail(generics.RetrieveDestroyAPIView):
```

Comment:
@SchrodingersGat any idea why the new API views won't work here? I am trying to run this and i always get:
'DELETE action unavailable at /api/notifications/3/'

``` python
constructForm(`/api/notifications/${$(this).attr('pk')}/`, {
        method: 'DELETE',
        title: '{% trans "Delete Notification" %}',
        onSuccess: function(data) {
            updateNotificationTables();
        }
    });
```

Reply:
Have you assigned the new `NotificationMessage` model to the permissions groups in `users.models`?

Reply:
Ah I see, it was in RULESET_IGNORE

Reply:
@SchrodingersGat am I allowed to add a new RULESET_MODELS category? I need something that every user is assigned.

Reply:
I thought about 'user' or 'base'

Reply:
If it was in `RULESET_IGNORE` then it *should* return "True" for any permission check:

https://github.com/inventree/InvenTree/blob/0cb66a45ae26274bfd711e8ac8017f7a0c6dfc2c/InvenTree/users/models.py#L207

Might need to trace through the code and see why the permissions checks are not running properly.

the API permissions are determined here:

https://github.com/inventree/InvenTree/blob/master/InvenTree/InvenTree/metadata.py

Reply:
@SchrodingersGat I think I found a flaw in the current metadata code that makes it that the metadata inclusion does not work. If a view does not provide a `POST`or `PUT` function `DELETE` and `GET`?  will also not work, you can see this when you create an Endpoint that inherits from `RetrieveDestroyAPIView` as this PR does.
A possible fix could be https://github.com/inventree/InvenTree/pull/2397 but I am not sure if I just misunderstand the functions of the new API forms. It is not the permissions, I used a superuser for all tests now.